blosc/blosc2.c

```diff
@@ -3100,7 +3176,9 @@ const char* blosc_get_compressor(void)
 
 int blosc_set_compressor(const char* compname) {
   int code = blosc_compname_to_compcode(compname);
-
+  if (code >= BLOSC_UDCODEC) {
```

Comment:
Please add a `BLOSC_TRACE_ERROR()` with the reason.  E.g. "User defined codecs cannot be set here.  Use Blosc2 mechanism instead."